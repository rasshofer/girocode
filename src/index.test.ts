import giroCode, { GiroCodeEncoding } from '.';

describe('giroCode', () => {
  it('works with sane values', () => {
    const result = giroCode({
      name: 'Wikimedia',
      iban: 'DE33100205000001194700',
      bic: 'BFSWDE33BER',
      amount: 1.23,
    });
    expect(result).toMatchSnapshot();
  });

  it('supports custom versions', () => {
    const result = giroCode({
      version: '001',
      name: 'Wikimedia',
      iban: 'DE33100205000001194700',
      bic: 'BFSWDE33BER',
      amount: 1.23,
    });
    expect(result).toMatchSnapshot();
  });

  it('supports custom encoding', () => {
    const result = giroCode({
      encoding: GiroCodeEncoding.ISO88591,
      name: 'Wikimedia',
      iban: 'DE33100205000001194700',
      bic: 'BFSWDE33BER',
      amount: 1.23,
    });
    expect(result).toMatchSnapshot();
  });

  it('supports custom currencies', () => {
    const result = giroCode({
      name: 'Wikimedia',
      iban: 'DE33100205000001194700',
      bic: 'BFSWDE33BER',
      amount: 4.56,
      currency: 'USD',
    });
    expect(result).toMatchSnapshot();
  });

  it('normalizes formatted IBANs', () => {
    const result = giroCode({
      name: 'Wikimedia',
      iban: 'DE33 1002 0500 0001 1947 00',
      bic: 'BFSWDE33BER',
      amount: 1.23,
    });
    expect(result).toMatchSnapshot();
  });

  it('normalizes text values (names, currencies, amounts, codes, references, purposes)', () => {
    const result = giroCode({
      name: 'A very very very very very very very very very very very very very very very very very very very very long name',
      iban: 'DE33     1002 0500 0001 1947 00',
      bic: 'BFSWDE33BER',
      amount: 1.23456,
      currency: 'EURO',
      code: 'ABCDEF',
      ref: 'A very very very very very very very very very very very very very very very very very very very very long reference',
      purpose:
        'A very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very long purpose',
    });
    expect(result).toMatchSnapshot();
  });

  it('detects and rejects invalid IBANs', () => {
    const result = giroCode({
      name: 'Wikimedia',
      iban: 'Some invalid IBAN',
      bic: 'BFSWDE33BER',
      amount: 1.23,
    });
    expect(result).toBeUndefined();
  });

  it('detects and rejects missing BICs if version 1 is used', () => {
    const result = giroCode({
      version: '001',
      name: 'Wikimedia',
      iban: 'DE33100205000001194700',
      amount: 1.23,
    });
    expect(result).toBeUndefined();
  });
});
