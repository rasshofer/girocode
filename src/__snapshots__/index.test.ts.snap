// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`giroCode normalizes formatted IBANs 1`] = `
"BCD
002
1
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
EUR1.23


"
`;

exports[`giroCode normalizes text values (names, currencies, amounts, codes, references, purposes) 1`] = `
"BCD
002
1
SCT
BFSWDE33BER
A very very very very very very very very very very very very very ver
DE33100205000001194700
EUR1.23
ABCD
A very very very very ver
A very very very very very very very very very very very very very very very very very very very very very very very very very very very ver"
`;

exports[`giroCode supports custom currencies 1`] = `
"BCD
002
1
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
USD4.56


"
`;

exports[`giroCode supports custom encoding 1`] = `
"BCD
002
2
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
EUR1.23


"
`;

exports[`giroCode supports custom versions 1`] = `
"BCD
001
1
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
EUR1.23


"
`;

exports[`giroCode supports empty amounts 1`] = `
"BCD
002
1
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
EUR0.00


"
`;

exports[`giroCode works with sane values 1`] = `
"BCD
002
1
SCT
BFSWDE33BER
Wikimedia
DE33100205000001194700
EUR1.23


"
`;
